packages/main/src/plugin/index.ts

```diff
@@ -99,6 +99,11 @@ import checkDiskSpace from 'check-disk-space';
 
 type LogType = 'log' | 'warn' | 'trace' | 'debug' | 'error';
 
+let isQuitting = false;
```

Comment:
Not sure to understand why we need an external function there and an attribute not attached to a class

PluginSystem instance can listen as well to the 'before-quit' event and just update a private field `isQuitting` then LoggerWithEnd will only do if `(!this.isQuitting)`

Reply:
Thanks for the suggestion. This hadn't occurred to me. I will move the `before-quit` to `PluginSystem` and add `isQuitting` as a private field.